### DISCUSSION TITLE
Feature: Coletar dados do Portal da Transparência

### DISCUSSION THREAD

**Discussion by augustresende** [BrasilAPI/BrasilAPI#221] [Ideas]
Estudar API do Portal da Transparência para ser implementado no Brasil API.

http://www.portaltransparencia.gov.br/api-de-dados

Um dos problemas possíveis é que existe um rate limit de 90 requisições por minuto das "6:00 às 23:59" e 300 requisições por minuto das "00:00 às 5:59". Uma alternativa é consultar diretamente das bases de dados, mas é necessário implementar alguma forma de mante-las disponíveis e atualizadas.

http://www.portaltransparencia.gov.br/download-de-dados

**Comment by hsfvieira** [comment]
Eu ia abrir uma Issue com exatamente o mesmo assunto, mas pensei a nível municipal, pois já tinha um projeto de coletar as informações da minha cidade e trazer de forma mais amigavel aos usuários.

Mas nesses casos, os sistemas são licitados, e cada empresa entrega os dados de uma determinada forma. Na minha cidade por exemplo eles não possuem uma rota para consulta externa, só posso extrair os dados filtrados através de CSV.

**Comment by elirweb** [comment]
Boa tarde,

@AugustoResende, se você ter um serviço web conectado com o "HangFire" ou outras ferramentas similares, para obter todos os dados necessarios de varias requisições ao mesmo tempo, que alimenta seu banco de dados  "arquivos json" ou sistema web. 
E na outra ponta possa disponibilizar um unico endpoint como se fosse um gateway para o usuario final possa informar o serviço e passa os dados para que ele consumir o seu serviço ao invés do "Portal da Transparência".
 E no seu serviço que integrado ao "Portal Transparência" tem um Job que fica batendo em torno de 5x por dia, para atualiazar sua massa de dados.
